### ISSUE TITLE
Channel Attention Module

### ISSUE THREAD

**Issue by devLupin** [devLupin/speech-emotion-recognition#2]
### Overview
- 특징의 채널 간 관계를 활용하여 channel attention map 생성
- channel attention은 입력 특징의 '무엇'이 의미가 있는지에 초점
- 입력 특징맵의 공간 차원 압축

### **Process**
- `average pooling`($F_{avg}$), `max pooling`($F_{max}$)
  - 두 개의 서로 다른 spatial context descriptor 생성
- shared MLP layer
  - channel attention map 생성
- $F_{avg}$, $F_{max}$ concat
- $sigmoid$

### computed as:
$\mathbf{M_{c}}\left(\mathbf{F}\right) = \sigma\left(\text{MLP}\left(\text{AvgPool}\left(\mathbf{F}\right)\right)+\text{MLP}\left(\text{MaxPool}\left(\mathbf{F}\right)\right)\right)$

### Figure
![image](https://user-images.githubusercontent.com/33558083/235321073-4a28f23f-a685-44a4-9e9d-a2d3db1a7a28.png)

### [Official code](https://github.com/Jongchan/attention-module/blob/5d3a54af0f6688bedca3f179593dff8da63e8274/MODELS/cbam.py#L26)(Pytorch)

**Comment by devLupin** [comment]
### Tensorflow 구현

```python
class ChannelGate(tf.keras.Model):

  def __init__(self, gate_channels):
    super(ChannelGate, self).__init__()
    self.gate_channels = gate_channels
    self.mlp = Sequential([
        Flatten(),
        Dense(gate_channels // 2, activation='relu'),
        Dense(gate_channels, activation=None)
    ])
  def call(self, x):
    channel_att_sum = None
    
    avg_pool_layer = AveragePooling2D(pool_size=(x.shape[1], x.shape[2]), strides=(x.shape[1], x.shape[2]))
    avg_pool = avg_pool_layer(x)
    avg_pool_raw = self.mlp(avg_pool)
    max_pool_layer = MaxPool2D(pool_size=(x.shape[1], x.shape[2]), strides=(x.shape[1], x.shape[2]))
    max_pool = max_pool_layer(x)
    max_pool_raw = self.mlp(max_pool)
    
    channel_att_sum = avg_pool_raw + max_pool_raw
    scale = tf.math.sigmoid(channel_att_sum)
    scale = tf.expand_dims(scale, axis=1)
    scale = tf.expand_dims(scale, axis=1)
    scale = tf.tile(scale, [1, x.shape[1], x.shape[2], 1])  # # expand_as in Pytorch
    
    return x * scale
```
